### PR TITLE
Update-roles

### PR DESCRIPTION
**Pull request by seanbsullivan**
Updating the roles index page and the associated role pages based on the current figma design

### CODE REVIEW

**Comment by seanbsullivan** [comment]
# Changes
- Updated the Roles page, added in images and text content from Figma, and updated the indiv Role page
- Noticed `RedButton` component became a white button with the new design :). Created a `WhiteButton`, made the change across all the affected files, and made `RedButton` red again, since the new design still has some red buttons (notably for me, on the cohort cards)

# TODO
- The spacing isn't quite right between the top element ("Pick the perfect role for you") on the roles index page (and the indiv role pages), and everything afterwards... if anyone (@wcrum @dfullaway) knows how to quickly fix that I'd appreciate any advice
- Once everything looks good on desktop, I'll work the mobile part. Big issue there will be that we need to take the alternating left/right stuff from the Roles index page and make it a column for mobile

**Comment by dfullaway** [comment]
The issue with spacing is because you are making the top section use the entire screen. You can swap the h-screen on line 17 of pages/roles/index.astro for something else (like h-fit) to shrink this down a bit. 
For the mobile/desktop page having different orders of elements, I solved a similar issue on the accordion by checking the width of the page, and determining if it was being viewed on mobile (width < 640 pixels). Then I could use if / then logic to swap elements in and out. You can look at the accordion branch in Accordion.svelte and AccordionSection.svelte to see how I did that (although there may be complications using that in an astro page). Also, I concur with William Crum's comment in the Signal chat, that it works a lot better to start with the mobile, and then move on to the desktop side.

**Comment by seanbsullivan** [comment]
@dfullaway Thanks for the info on h-fit! I knew it was something related to h-screen but I'm not super experienced with frontend stuff so that's great to have something to look into for fixing that issue. 

I had the same thought regarding checking if we're being viewed on mobile and adjusting the order based on that, good to hear that's the route you went with the accordion and I'll take a look at that code you mentioned. 

Appreciate the feedback on starting with mobile first and as I keep working on these pages I'll do it with the mobile view first. Probably won't get much more done this weekend with it but next week will hopefully have time to dive back in again. Thanks again for all the feedback.

File: src/content/roles/prod_designer.md [hunk 1]
```diff
@@ -2,15 +2,21 @@
 title: 'ABOUT THE JOB'
```
Comment (dfullaway) [inline src/content/roles/prod_designer.md]: All of the roles have a title of  'ABOUT  THE  JOB'. Consider  removing it from the roles and just adding the text to the  page as needed.

File: src/pages/roles/index.astro [hunk 2]
```diff
@@ -2,63 +2,88 @@
...
-                  <div class="w-1/2 p-4">
-                    <Hero
-                      title={post.frontmatter.role}
-                      content={post.frontmatter.about}
-                      link={"/roles/" + post.frontmatter.url}
-                      padding="pt-48"
-                      linkText="LEARN MORE"
-                    />
+    <div
+      class="relative z-20 h-fit snap-start pt-20 pb-20 bg-[url('/images/group_of_marines.png')] bg-no-repeat"
```
Comment (dfullaway) [inline src/pages/roles/index.astro]: Background image does  not  scale properly (on widescreen, the image is pushed  to the left with blank space to the right). Adding bg-cover should correct the issue.

File: src/pages/roles/index.astro [hunk 3]
```diff
@@ -2,63 +2,88 @@
...
+      />
+    </div>
+    <div class="max-h-screen scrollbar-hide snap-start snap-y snap-mandatory">
+      {
+        posts.map((post) => {
+          if (side == 'left') {
+            side = 'right'
+            return (
+              <>
+                <div class="h-screen snap-start py-20 px-10">
```
Comment (dfullaway) [inline src/pages/roles/index.astro]: You can use the ContentImageFull elements to  make this much more compact and align with what has been done elsewhere on the site. It is also already setup to work on mobile, and the only change between cards is whether to include "flipped" or not.

File: src/pages/roles/index.astro [hunk 4]
```diff
@@ -2,63 +2,88 @@
...
+import FooterFull from '../../components/FooterFull.svelte'
 const posts = await Astro.glob('../../content/roles/*.md')
 let side = 'left'
 ---
 
-<Layout title="ABOUT">
+<Layout title="Roles">
   <div
-    class="max-h-screen overflow-y-scroll scrollbar-hide snap-start snap-y snap-mandatory"
+    class="relative max-h-screen overflow-y-scroll font-expressway red-radial-gradient scrollbar-hide snap-start snap-y snap-mandatory"
```
Comment (dfullaway) [inline src/pages/roles/index.astro]: Add EGA to background. (See root index.astro  lines 17-19.)

File: src/pages/roles/[...role].astro [hunk 5]
```diff
@@ -18,38 +20,33 @@ const { frontmatter, compiledContent } = Astro.props
 const content = compiledContent()
 ---
 
-<Layout title="">
+<Layout title={frontmatter.role}>
   <div
-    class="max-h-screen overflow-y-scroll scrollbar-hide snap-start snap-y snap-mandatory"
+    class="relative max-h-fit overflow-y-scroll font-expressway red-radial-gradient scrollbar-hide snap-start snap-y snap-mandatory"
```
Comment (dfullaway) [inline src/pages/roles/[...role].astro]: Add EGA to background. (See root index.astro lines 17-19.)

File: src/pages/roles/[...role].astro [hunk 6]
```diff
@@ -18,38 +20,33 @@ const { frontmatter, compiledContent } = Astro.props
...
   >
-    <div class="h-screen snap-start py-20">
+    <div
+      class="relative z-20 h-fit snap-start pt-20 pb-0 bg-no-repeat"
```
Comment (dfullaway) [inline src/pages/roles/[...role].astro]: Background image does not scale properly (on widescreen, the image is pushed to the left with blank space to the right). Adding bg-cover should correct the issue.

File: src/pages/roles/[...role].astro [hunk 7]
```diff
@@ -18,38 +20,33 @@ const { frontmatter, compiledContent } = Astro.props
...
+      style={{
+        backgroundImage: `url('/images/${frontmatter.image_wide}')`,
+      }}
+    >
       <Hero
```
Comment (dfullaway) [inline src/pages/roles/[...role].astro:37]: The default padding on the Hero element is excessive here. Consider adding a padding parameter here.

File: src/pages/roles/[...role].astro [hunk 8]
```diff
@@ -4,6 +4,8 @@ import Hero from '../../components/elements/Hero.astro'
 import Card from './RoleCard.astro'
```
Comment (dfullaway) [inline src/pages/roles/[...role].astro]: This is not used, and should be removed.

File: src/pages/roles/RoleCard.astro
Comment (dfullaway) [inline src/pages/roles/RoleCard.astro:1]: This is never used. Not sure what it was supposed to be - name indicates it should be a card with a role on it, but it looks like a cohort card.

File: src/components/sections/Vision.astro [hunk 9]
```diff
@@ -1,9 +1,8 @@
 ---
 import Hero from '../elements/Hero.astro'
```
Comment (dfullaway) [inline src/components/sections/Vision.astro]: Not used. Remove.

File: src/components/sections/Vision.astro
Comment (dfullaway) [inline src/components/sections/Vision.astro:1]: I don't see where this is used. Looks like a hold over from earlier about page.

File: src/components/Accordion.svelte
Comment (dfullaway) [inline src/components/Accordion.svelte:1]: We need to be careful in merging pull requests - we don't  want this overwriting the accordion work in the other PR.

**Review by dfullaway — COMMENTED** [review]
I've put in a few suggestions to correct a couple of issues and  hopefully make the mobile work a bit easier.

File: src/content/roles/prod_designer.md (hunk 1 above)
Comment (seanbsullivan) [reply src/content/roles/prod_designer.md]: Thanks and yeah looking at that it doesn't appear that anything was using that element from the role files anyways so I took it out of all of them

File: src/pages/roles/index.astro (hunk 3 above)
Comment (seanbsullivan) [reply src/pages/roles/index.astro]: The only issue I see with using ContentImageFull here is that there's no property that would let me set the src for the image. Do you think I should modify ContentImageFull to add that property? Right now it's just hard coded to use a dummy image

File: src/components/Accordion.svelte
Comment (seanbsullivan) [reply src/components/Accordion.svelte:1]: Definitely, good catch. I'll leave this as unresolved as a reminder to make sure the changes from your accordion work are merged in before this PR is merged

**Pull request merged**